pyro/primitives.py

```diff
@@ -148,6 +148,40 @@ def deterministic(name, value, event_dim=None):
     return sample(name, dist.Delta(value, event_dim=event_dim).mask(False), obs=value)
 
 
+@effectful(type="subsample")
+def subsample(data, event_dim):
+    """
+    EXPERIMENTAL Subsampling statement to subsample data based on enclosing
```

Comment:
nit: make this a Sphinx warning?

Reply:
This is how we mark experimental everywhere. I can change that convention, but we should do so in a separate PR that changes all relevant docstrings.